File: MatlabOctaveCodeSnippets/snip_signals_three_rects.m
%% Mimicking a continuous-time signal by using plot
Ts = 0.001; %sampling interval in seconds
t = -1.6:Ts:1.6; %discrete-time in seconds
pulse_width = 0.2; %support of this rect in seconds
x = rectpuls(t,pulse_width) - 3*rectpuls(t-0.2,pulse_width) + ...
    3*rectpuls(t-0.4,pulse_width); %define the signal
plot(t,x) %plot as a continuos-time signal
xlabel('t'), ylabel('x(t)')

File: PythonCodeSnippets/snip_signals_three_rects.py
import numpy as np
import matplotlib.pyplot as plt

def rectpuls(t, width):
    """Generates a rectangular pulse with specified width."""
    return np.where(np.abs(t) <= width / 2, 1, 0)

# Sampling interval in seconds
Ts = 0.001
# Discrete time in seconds
t = np.arange(-1.6, 1.6, Ts)
# Support width of this rectangle in seconds
pulse_width = 0.2
# Defines the signal
x = (
    rectpuls(t, pulse_width)
    - 3 * rectpuls(t - 0.2, pulse_width)
    + 3 * rectpuls(t - 0.4, pulse_width)
)
# Plots as a continuous signal
plt.plot(t, x)
plt.xlabel("t")
plt.ylabel("x(t)")
plt.show()
